includes/class-wc-payments-incentives-service.php

```diff
@@ -0,0 +1,208 @@
+		global $menu;
+
+		if ( ! $this->get_cached_connect_incentive() ) {
+			return;
+		}
+
+		foreach ( $menu as $index => $menu_item ) {
+			if ( 'wc-admin&path=/payments/connect' === $menu_item[2] ) {
+				$menu[ $index ][0] .= WC_Payments_Admin::MENU_NOTIFICATION_BADGE; // phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
+				break;
+			}
+		}
+	}
+
+	/**
+	 * Adds allowed promo notes from eligible incentive.
+	 *
+	 * @param array $promo_notes Current allowed promo notes.
+	 * @return array Updated allowed promo notes.
+	 */
+	public function allowed_promo_notes( $promo_notes = [] ): array {
+		if ( $this->get_cached_connect_incentive() ) {
+			/**
+			 * Suppress psalm error because we already check that `id` exists in `is_valid_cached_incentive`.
+			 *
+			 * @psalm-suppress PossiblyNullArrayAccess */
+			$promo_notes[] = $this->get_cached_connect_incentive()['id'];
+		}
+
+		return $promo_notes;
+	}
+
+	/**
+	 * Gets and caches eligible connect incentive from the server.
+	 *
+	 * @param bool $force_refresh Forces data to be fetched from the server, rather than using the cache.
+	 *
+	 * @return array|null List of incentives or null.
+	 */
+	public function get_cached_connect_incentive( bool $force_refresh = false ): ?array {
+		// Return early if there is an account connected.
+		if ( WC_Payments::get_account_service()->is_stripe_connected() ) {
+			return null;
+		}
+
+		// Return early if the country is not supported.
+		if ( ! array_key_exists( WC()->countries->get_base_country(), WC_Payments_Utils::supported_countries() ) ) {
+			return null;
+		}
+
+		$incentive = $this->database_cache->get_or_add(
+			Database_Cache::CONNECT_INCENTIVE_KEY,
+			[ $this, 'fetch_connect_incentive' ],
+			'is_array',
+			$force_refresh,
+		);
+
+		if ( ! $this->is_valid_cached_incentive( $incentive ) ) {
+			return null;
+		}
+
+		return $incentive;
+	}
+
+	/**
+	 * Fetches eligible connect incentive from the server.
+	 *
+	 * @return array|null Array of eligible incentive data or null.
+	 */
+	public function fetch_connect_incentive(): ?array {
+		// Request incentive from WCPAY API.
+		$url = add_query_arg(
+			[
+				// Store ISO-2 country code, e.g. `US`.
+				'country'      => WC()->countries->get_base_country(),
+				// Store locale, e.g. `en_US`.
+				'locale'       => get_locale(),
+				// WCPay active for duration in seconds.
+				'active_for'   => time() - get_option( 'wcpay_activation_timestamp', time() ),
+				// Whether the store has paid orders in the last 90 days.
+				'has_orders'   => ! empty(
+					wc_get_orders(
+						[
+							'status'       => [ 'wc-completed', 'wc-processing' ],
+							'date_created' => '>=' . strtotime( '-90 days' ),
+							'return'       => 'ids',
+							'limit'        => 1,
+						]
+					)
+				),
+				// Whether the store has at least one payment gateway enabled.
+				'has_payments' => ! empty( WC()->payment_gateways()->get_available_payment_gateways() ),
```

Comment:
I wonder if we should be sending `has_wcpay` also with a trimmed-down logic since the plugin is obviously active. I am especially looking at the orders processed with WCPay signal. I guess we could deduce that on the server via transactions, but not if the merchant stopped using WCPay at some point.

Reply:
We should bear in mind that coming from WCPay, `has_payments` means that there is another payments gateway enabled, not WCPay, since if WCPay is not connected or payments enabled on the account, the payment gateway `is_available` [method](https://github.com/Automattic/woocommerce-payments/blob/0f03a5f3c5e497b6bf68a7201dd0e830f6881ff5/includes/class-wc-payment-gateway-wcpay.php#L488-L488) will return `false` and WCPay will not be returned by `get_available_payment_gateways`.

Just something for us to keep in mind so as not to assume this will always be `true` since WCPay is obviously active (that was my initial thought).

Reply:
>I wonder if we should be sending has_wcpay also with a trimmed-down logic since the plugin is obviously active.

This is a similar case to `active_for`, I didn't want to change the meaning of `has_wcpay`. I think we have pretty clear that it's true by just having WCPay installed.

>I am especially looking at the orders processed with WCPay signal. I guess we could deduce that on the server via transactions.

I don't see how it can happen, given the client conditions, every call to the endpoint will be without a WCPay account, so no way for use to check anything about a previous account.

>We should bear in mind that coming from WCPay, has_payments...

Yep, I'm aware of it, that's why I kept it as it is. Because we want to know if there is any other payment method, except for WCPay.